File: etamen-backend/app/Modules/Radiology/Infrastructure/Models/RadiologyScanCategory.php
<?php

namespace App\Modules\Radiology\Infrastructure\Models;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class RadiologyScanCategory extends Model
{
    protected $fillable = [
        'code',
        'name_ar',
        'name_en',
        'description_ar',
        'description_en',
        'is_active',
        'sort_order',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'sort_order' => 'integer',
        ];
    }

    public function scopeActive(Builder $query): Builder
    {
        return $query->where('is_active', true);
    }

    public function scans(): HasMany
    {
        return $this->hasMany(RadiologyScan::class);
    }

    public function instructions(): HasMany
    {
        return $this->hasMany(RadiologyPreparationInstruction::class);
    }
}

File: etamen-flutter/lib/features/radiology/domain/entities/radiology_scan_category.dart
class RadiologyScanCategory {
  const RadiologyScanCategory({
    required this.id,
    required this.code,
    required this.nameAr,
    this.nameEn,
    this.descriptionAr,
    this.descriptionEn,
    this.isActive = true,
    this.sortOrder = 0,
  });

  final int id;
  final String code;
  final String nameAr;
  final String? nameEn;
  final String? descriptionAr;
  final String? descriptionEn;
  final bool isActive;
  final int sortOrder;

  String name(bool isArabic) {
    if (!isArabic && nameEn?.trim().isNotEmpty == true) return nameEn!.trim();
    return nameAr;
  }

  String? description(bool isArabic) {
    if (!isArabic && descriptionEn?.trim().isNotEmpty == true) {
      return descriptionEn!.trim();
    }
    return descriptionAr;
  }
}
